warp10/src/main/java/io/warp10/continuum/gts/GTSHelper.java

```diff
@@ -1897,14 +1897,64 @@ public static final GeoTimeSerie subSerie(GeoTimeSerie gts, long starttimestamp,
     // Extract values/locations/elevations that lie in the requested interval
     //
 
-    // We know how many data will the new GTS so we provision arrays to receive the data.
+    // We know how many data the new GTS will have so we provision arrays to receive the data.
     int count = lastidx - firstidx + 1;
-    GTSHelper.multiProvision(subgts, gts.type, count, count);
 
-    for (int i = firstidx; i <= lastidx; i++) {
-      setValue(subgts, gts.ticks[i], null != gts.locations ? gts.locations[i] : GeoTimeSerie.NO_LOCATION, null != gts.elevations ? gts.elevations[i] : GeoTimeSerie.NO_ELEVATION, valueAtIndex(gts, i), overwrite);
+    // Allocate locations/elevations arrays so they get expanded by the call to multiProvision
+    if (null != gts.locations) {
+      subgts.locations = new long[0];
+    }
+    if (null != gts.elevations) {
+      subgts.elevations = new long[0];
     }
 
+    GTSHelper.multiProvision(subgts, gts.type, count, count);
+
+    if (!overwrite) {
+      if (null != subgts.locations) {
+        System.arraycopy(gts.locations, firstidx, subgts.locations, 0, count);
```

Comment:
In fact, locations and elevations have not been expanded by multiProvision, so this gives an ArrayIndexOutOfBoundsException